lib/luaby/ast/anonymous_function.rb

```diff
@@ -0,0 +1,12 @@
+class Luaby::AST::AnonymousFunction
+  attr_reader :params, :body
+
+  def initialize(params, body)
+    @params = params
+    @body = body
+  end
+
+  def to_lua
+    "function (#{params.to_lua})\n#{body.to_lua.lines.map { |x| "#{Luaby::INDENT}#{x}" }.join}\nend"
```

Comment:
If `body` is a `AST::Function` rather than an `AST::Block` then this could be `"function #{body.to_lua}"`